pennylane/pauli/dla/lie_closure.py

```diff
@@ -13,59 +13,64 @@
 # limitations under the License.
 """A function to compute the Lie closure of a set of operators"""
 import warnings
-from collections.abc import Iterable
 from copy import copy
 from functools import reduce
 
 # pylint: disable=too-many-arguments
 from itertools import product
-from typing import Union
+from typing import Iterable, Union
 
 import numpy as np
 import scipy
 
 import pennylane as qml
 from pennylane.operation import Operator
+from pennylane.typing import TensorLike
 
 from ..pauli_arithmetic import PauliSentence, PauliWord
+from ..trace_inner_product import trace_inner_product
 
 
 def lie_closure(
-    generators: Iterable[Union[PauliWord, PauliSentence, Operator]],
+    generators: Iterable[Union[PauliWord, PauliSentence, Operator, TensorLike]],
+    *,  # force non-positional kwargs of the following
     max_iterations: int = 10000,
     verbose: bool = False,
     pauli: bool = False,
+    matrix: bool = False,
```

Comment:
```suggestion
    use_matrix_rep: bool = False,
```
I think we may want to be a bit more clear here. My intuition when seeing `matrix` as an argument is that the function expects an actual matrix, not a boolean. Maybe something like `use_matrix_rep` would clarify things.
Similar concerns for `pauli`.

Reply:
note that `pauli` as a keyword like this is not newlt introduced here and already part of the UI for a while, and also used in several other places, e.g. [qml.dot](https://docs.pennylane.ai/en/stable/code/api/pennylane.dot.html), so I wouldnt make a breaking change here.

if product prefers `use_matrix_rep` i can make the change to that 👍

Reply:
@Qottmann quickly discussed it, leaving it as `matrix` is fine 👌